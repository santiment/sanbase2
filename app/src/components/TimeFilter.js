import React from 'react'
import cx from 'classnames'
import './TimeFilter.css'

const timeOptionsDefault = ['1w', '1m', '3m', '6m', '1y', 'all']

export const TimeFilterItem = ({
  isSelected = false,
  value = 'all',
  setFilter,
  disabled = false
}) => (
  <div
    className={cx({
      'time-filter-item': true,
      'time-filter-item--selected': isSelected,
      'time-filter-item--disabled': disabled
    })}
    onClick={() => !disabled && setFilter(value)}
  >
    {value}
  </div>
)

export const TimeFilter = ({
  selected = 'all',
  timeOptions = timeOptionsDefault,
  setFilter,
  disabled = false
}) => (
  <div className='time-filter'>
    {timeOptions.map(option => (
      <TimeFilterItem
        key={option}
        isSelected={selected === option}
        value={option}
        setFilter={setFilter}
        disabled={disabled}
      />
    ))}
  </div>
)

export default TimeFilter
