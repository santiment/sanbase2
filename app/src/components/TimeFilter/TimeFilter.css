.time-filter {
  display: flex;
  user-select: none;
  cursor: pointer;
  min-width: 32px;
  text-align: center;
  border-radius: 4px;
  padding: 10px 25px;
}

.time-filter-item {
  cursor: pointer;
  min-width: 32px;
  text-align: center;
  border-radius: 4px;
}

.time-filter .time-filter-item--selected {
  background: rgba(38, 43, 51, 0.8);
  color: white !important;
}

.time-filter .time-filter-item--disabled {
  color: #888;
  cursor: not-allowed;
}

.night-mode .time-filter,
.night-mode .time-filter-item--selected {
  color: #a0a0a0;
  background-color: #252932;
}

.night-mode .time-filter-item--disabled {
  color: #595959;
}
