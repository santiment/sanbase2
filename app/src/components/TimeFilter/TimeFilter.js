import React, { Component } from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import './TimeFilter.css'

const timeOptionsDefault = ['1w', '1m', '3m', '6m', '1y', 'all']

export const TimeFilterItem = ({
  isSelected = false,
  value = 'all',
  setFilter,
  disabled = false
}) => (
  <div
    className={cx({
      'time-filter-item': true,
      'time-filter-item--selected': isSelected,
      'time-filter-item--disabled': disabled
    })}
    onClick={() => !disabled && setFilter(value)}
  >
    {value}
  </div>
)

class TimeFilter extends Component {
  state = {
    selected: this.props.defaultSelected
  }

  static defaultProps = {
    defaultSelected: 'all',
    timeOptions: timeOptionsDefault,
    disabled: false
  }

  static propTypes = {
    defaultSelected: PropTypes.string,
    timeOptions: PropTypes.array,
    onSelectOption: PropTypes.func,
    disabled: PropTypes.bool
  }

  onSelectOption = newOption => {
    this.setState({ selected: newOption }, () => {
      this.props.onSelectOption(newOption)
    })
  }

  render () {
    const { selected } = this.state
    const { timeOptions, disabled } = this.props
    return (
      <div className='time-filter'>
        {timeOptions.map(option => (
          <TimeFilterItem
            key={option}
            isSelected={selected === option}
            value={option}
            setFilter={this.onSelectOption}
            disabled={disabled}
          />
        ))}
      </div>
    )
  }
}

export default TimeFilter
