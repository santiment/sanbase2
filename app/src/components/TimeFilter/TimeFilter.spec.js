/* eslint-env jest */
import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import TimeFilter, { TimeFilterItem } from './TimeFilter'

describe('TimeFilter component', () => {
  it('smoke', () => {
    const mockCb = jest.fn()
    const wrapper = shallow(<TimeFilter onSelectOption={mockCb} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should be called onSelectOption, if TimeFilter option is clicked', () => {
    const mockCb = jest.fn()
    const wrapper = mount(<TimeFilter onSelectOption={mockCb} />)
    const firstTimeFilterItem = wrapper.find('.time-filter-item').at(2)
    firstTimeFilterItem.simulate('click')
    expect(mockCb).toHaveBeenCalled()
  })

  it('should change selected option, after click on not selected item', () => {
    const mockCb = jest.fn()
    const wrapper = mount(<TimeFilter onSelectOption={mockCb} />)
    expect(
      wrapper
        .find(TimeFilterItem)
        .at(2)
        .props().isSelected
    ).toBeFalsy()
    expect(
      wrapper
        .find('.time-filter-item')
        .at(2)
        .text()
    ).toBe('3m')
    const firstTimeFilterItem = wrapper.find('.time-filter-item').at(2)
    firstTimeFilterItem.simulate('click')
    expect(wrapper.state('selected')).toBe('3m')
  })

  it('should have selected option is 3m, if defaultSelected is 3m', () => {
    const wrapper = shallow(<TimeFilter defaultSelected='3m' />)
    expect(
      wrapper
        .find(TimeFilterItem)
        .at(2)
        .props().isSelected
    ).toBeTruthy()
  })
})
