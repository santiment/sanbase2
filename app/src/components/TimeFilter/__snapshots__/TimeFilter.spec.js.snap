// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeFilter component smoke 1`] = `
<div
  className="time-filter"
>
  <Component
    disabled={false}
    isSelected={false}
    key="1w"
    setFilter={[Function]}
    value="1w"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="1m"
    setFilter={[Function]}
    value="1m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="3m"
    setFilter={[Function]}
    value="3m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="6m"
    setFilter={[Function]}
    value="6m"
  />
  <Component
    disabled={false}
    isSelected={false}
    key="1y"
    setFilter={[Function]}
    value="1y"
  />
  <Component
    disabled={false}
    isSelected={true}
    key="all"
    setFilter={[Function]}
    value="all"
  />
</div>
`;
